### ISSUE TITLE
Spaces around chapter-verse-separator should not be ignored

### ISSUE THREAD

**Issue by renehamburger** [openbibleinfo/Bible-Passage-Reference-Parser#38]
When the comma is used as chapter-verse separator, as is the case in German Bible references, for example, it can easily be confused with the comma as a list-item separator. The universal practice is that the comma is followed by a space to signify the latter (e.g. `Matt 1, 3` ) and that a comma without a space is used to signify the former (e.g. `Matt 1,3`).

The parser currently ignores any number of spaces before or after the chapter-verse-separator, which leads to completely wrong interpretations of a list of chapters or verses separated with `, `. For example, `Matt 1, 3-4, 6` refers to 'Matt.1,Matt.3-Matt.4,Matt.6' but is currently parsed as 'Matt.1.3-Matt.4.5'.

Is this something you would be able to fix?

**Comment by alerque** [comment]
I don't suppose it's any use asking all German publishers past and future to use a less ambiguous separator and stop depending only on white-space for semantic meaning? :smirk:

**Comment by renehamburger** [comment]
Yes, it's very unfortunate that this separator is so ambiguous! I'm sure it has a very long heritage.
I've resolved this issue downstream now by converting those instances to ';' before passing them to the parser. I'd still leave this issue open as it would be great to resolve it here. (I personally found that part of this library too complicated to try to do it myself.)

But I would like to update the [definitions](https://github.com/openbibleinfo/Bible-Passage-Reference-Parser/edit/master/src/de/data.txt) for German references. Terms like 'Vers' (verse) and 'Kapitel' (chapter) need to be added with all possible case-number endings to work reliably.

@openbibleinfo, would you be prepared to accept such a PR?

**Comment by renehamburger** [comment]
I've created the PR: #40.

**Comment by openbibleinfo** [comment]
@renehamburger That is indeed very specific. I don't think I'll adjust the parser to handle this case since, as in #57, this solution would solve this specific case but break the general case for people who aren't as precise in their expression.

If you wanted, you could potentially get around this by loosening the `passage_existence_strategy` to just `b`, getting the full strings you're interested in, replacing ", " with ":", and then using the `punctuation_strategy` = `us`-style parser to grab the references. It's a bit hacky.

You could also potentially adjust the `.pegjs` grammar for German to allow this approach. The automated build process would overwrite any manual changes you make, however, so that would also have to be adjusted.

The longer term solution would be to allow adding custom matchers at runtime, but I'm not sure how possible that that is, given that the `.pegjs` files are built at compile time. I added it as a note in the backlog in the readme.

**Comment by openbibleinfo** [comment]
This functionality is now available in 3.1.0-alpha. Here's how to get what you describe:

```javascript
bcv.set_options({punctuation_strategy: "eu"});
bcv.set_options({
  grammar: {
    cv_sep_eu: /^,(?!\s)/,
    c_sep_eu: /^,\s+/
  }
});
bcv.parse("Matt 1, 3").osis(); // Matt.1,Matt.3
bcv.parse("Matt 1,3").osis(); // Matt.1.3
bcv.parse("Matt 1, 3-4, 6").osis(); // Matt.1,Matt.3-Matt.4,Matt.6
```

These specific values also have tests for them, so they shouldn't regress. I tried to test some fairly complex strings, but I'm not as familiar with how they should all work as you are.
